File: backend/src/ai/objects/apiTypes/AISearchResponse.scala
package delivery.ai.objects.apiTypes

final case class AIRecommendedProduct(
    productId: String,
    productName: String,
    price: Double,
    reason: String
)

final case class AIRecommendedMerchant(
    merchantId: String,
    storeName: String,
    category: String,
    reason: String,
    products: List[AIRecommendedProduct]
)

final case class AISearchResponse(
    query: String,
    merchants: List[AIRecommendedMerchant],
    summary: String
)

File: frontend/src/objects/ai/apiTypes/AISearchResponse.ts
export interface AIRecommendedProduct {
  productId: string
  productName: string
  price: number
  reason: string
}

export interface AIRecommendedMerchant {
  merchantId: string
  storeName: string
  category: string
  reason: string
  products: AIRecommendedProduct[]
}

export interface AISearchResponse {
  query: string
  merchants: AIRecommendedMerchant[]
  summary: string
}
